src/banman.cpp

```diff
@@ -129,22 +109,18 @@ void BanMan::Ban(const CSubNet& sub_net, int64_t ban_time_offset, bool since_uni
 
     {
         LOCK(m_cs_banned);
+        // If ban entry is absent, add it. If it's shorter, update it.
         if (m_banned[sub_net].nBanUntil < ban_entry.nBanUntil) {
             m_banned[sub_net] = ban_entry;
             m_is_dirty = true;
         } else
-            return;
+            return false; // nothing to do, longer ban already exists
```

Comment:
877b14aaeb ISTM this behavior ought to be documented/clarified in the setban rpc help

Reply:
Done.